javalib/src/main/scala/java/nio/GenHeapBuffer.scala

```diff
@@ -100,7 +100,13 @@ private[nio] final class GenHeapBuffer[B <: Buffer](val self: B)
 
     val len = remaining()
     System
-      .arraycopy(_array, _arrayOffset + position(), _array, _arrayOffset, len)
+      .arraycopy(
+        _array.toArray(),
```

Comment:
We probably should make _array.toArray() a val, otherwise we risk creation and copying of two arrys